neural_structured_learning/examples/notebooks/adversarial_cnn_transfer_learning_fashionmnist.ipynb

```diff
@@ -0,0 +1,1263 @@
+        "\n",
+        "![Adversarial Example](https://i.imgur.com/FyYq2Q0.png)\n",
+        "<font size=2>Source: [Explaining and Harnessing Adversarial Examples, Goodfellow et al., 2014](https://arxiv.org/abs/1412.6572)</font>\n",
+        "\n",
+        "Here, starting with the image of a panda, the attacker adds small perturbations (distortions) to the original image, which results in the model labelling this image as a gibbon, with high confidence. The process of adding these perturbations is explained below.\n",
+        "\n",
+        "## Fast gradient sign method\n",
+        "\n",
+        "The fast gradient sign method works by using the gradients of the neural network to create an adversarial example. For an input image, the method uses the gradients of the loss with respect to the input image to create a new image that maximises the loss. This new image is called the adversarial image. This can be summarised using the following expression:\n",
+        "$$adv\\_x = x + \\epsilon*\\text{sign}(\\nabla_xJ(\\theta, x, y))$$\n",
+        "\n",
+        "where \n",
+        "\n",
+        "*   adv_x : Adversarial image.\n",
+        "*   x : Original input image.\n",
+        "*   y : Original input label.\n",
+        "*   $\\epsilon$ : Multiplier to ensure the perturbations are small.\n",
+        "*   $\\theta$ : Model parameters.\n",
+        "*   $J$ : Loss.\n",
+        "\n",
+        "The gradients are taken with respect to the input image because the objective is to create an image that maximizes the loss. A method to accomplish this is to find how much each pixel in the image contributes to the loss value, and add a perturbation accordingly. This works pretty fast because it is easy to find how much each input pixel contributes to the loss by using the chain rule and finding the required gradients. Since our goal here is to attack a model that has already been trained, the gradient is not taken with respect to the trainable variables, i.e., the model parameters, which are now frozen.\n",
+        "\n",
+        "So let's try and fool our pretrained VGG19 model."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "hvIozwLu5rV5"
+      },
+      "source": [
+        "## Utility Functions for FGSM\n",
+        "\n",
+        "1. __`get_model_preds(...)`__: Helps in getting the top predicted class label and probability of an input image based on a specific trained CNN model\n",
+        "\n",
+        "2. __`generate_adverarial_pattern(...)`__: Helps in getting the gradients and the sign of the gradients w.r.t the input image and the trained CNN model\n",
```

Comment:
s/adverarial/adversarial, here and everywhere else in the tutorial where this typo exists.

Reply:
sure done